matsim/src/main/java/ch/sbb/matsim/routing/pt/raptor/SwissRailRaptorData.java

```diff
@@ -42,547 +42,603 @@
+	final Map<String, Map<String, QuadTree<TransitStopFacility>>> stopFilterAttribute2Value2StopsQT;
+	final OccupancyData occupancyData;
+
+	private SwissRailRaptorData(RaptorStaticConfig config, int countStops, RRoute[] routes, int[] departures,
+			Vehicle[] departureVehicles, Id<Departure>[] departureIds, RRouteStop[] routeStops, RTransfer[] transfers,
+			Map<TransitStopFacility, Integer> stopFacilityIndices,
+			Map<TransitStopFacility, int[]> routeStopsPerStopFacility, QuadTree<TransitStopFacility> stopsQT,
+			OccupancyData occupancyData) {
+		this.config = config;
+		this.countStops = countStops;
+		this.countRouteStops = routeStops.length;
+		this.routes = routes;
+		this.departures = departures;
+		this.departureVehicles = departureVehicles;
+		this.departureIds = departureIds;
+		this.routeStops = routeStops;
+		this.transfers = transfers;
+		this.stopFacilityIndices = stopFacilityIndices;
+		this.routeStopsPerStopFacility = routeStopsPerStopFacility;
+		this.stopsQT = stopsQT;
+		this.stopFilterAttribute2Value2StopsQT = new HashMap<>();
+		this.occupancyData = occupancyData;
+	}
+
+	public static SwissRailRaptorData create(TransitSchedule schedule, @Nullable Vehicles transitVehicles,
+			RaptorStaticConfig staticConfig, Network network, OccupancyData occupancyData) {
+		log.info("Preparing data for SwissRailRaptor...");
+		long startMillis = System.currentTimeMillis();
+
+		Map<Id<Vehicle>, Vehicle> vehicles = transitVehicles == null ? Collections.emptyMap()
+				: transitVehicles.getVehicles();
+		int countRoutes = 0;
+		long countRouteStops = 0;
+		long countDepartures = 0;
+
+		for (TransitLine line : schedule.getTransitLines().values()) {
+			countRoutes += line.getRoutes().size();
+			for (TransitRoute route : line.getRoutes().values()) {
+				countRouteStops += route.getStops().size();
+				countDepartures += route.getDepartures().size();
+			}
+		}
+
+		if (countRouteStops > Integer.MAX_VALUE) {
+			throw new RuntimeException("TransitSchedule has too many TransitRouteStops: " + countRouteStops);
+		}
+		if (countDepartures > Integer.MAX_VALUE) {
+			throw new RuntimeException("TransitSchedule has too many Departures: " + countDepartures);
+		}
+
+		int[] departures = new int[(int) countDepartures];
+		Vehicle[] departureVehicles = new Vehicle[(int) countDepartures];
+		Id<Departure>[] departureIds = new Id[(int) countDepartures];
+		RRoute[] routes = new RRoute[countRoutes];
+		RRouteStop[] routeStops = new RRouteStop[(int) countRouteStops];
+
+		int indexRoutes = 0;
+		int indexRouteStops = 0;
+		int indexDeparture = 0;
+
+		// enumerate TransitStopFacilities along their usage in transit routes to
+		// (hopefully) achieve a better memory locality
+		// well, I'm not even sure how often we'll need the transit stop facilities,
+		// likely we'll use RouteStops more often
+		Map<TransitStopFacility, Integer> stopFacilityIndices = new HashMap<>(
+				(int) (schedule.getFacilities().size() * 1.5));
```

Comment:
for the future: If you have a need to reformat, please commit this separately. This way, it's very hard to see what has actually changed in the file.